src/dra/63ED4.c

```diff
@@ -602,7 +602,105 @@ void func_8010427C(void) {
     FreePrimitives(D_80137E48);
 }
 
-INCLUDE_ASM("dra/nonmatchings/63ED4", func_801042C4);
+void func_801042C4(s32 arg0) {
+    VECTOR sp10;
+    Primitive* prim;
+    s32 i;
+
+    D_80137EE0 = arg0;
+    D_80097C98 = 0;
+    func_800EA5E4(0x18U);
+    if (arg0 == 1) {
+        func_800EA5E4(0x19U);
+        func_800EA5E4(0x401EU);
+    }
+    D_801379BC = 0;
+    D_80137E54 = 0;
+    PlaySfx(0xA);
+    D_80097928 = 1;
+    func_80103EAC();
+    D_801379B8 = ((s32)(g_StageId & 0x20) / 2) + 8;
+    D_801379C8.vx = D_801379C8.vy = D_801379C8.vz = 0;
+    D_801379C0.vx = D_801379C0.vy = D_801379C0.vz = 0;
+
+    D_801379D0.vx = D_801379D0.vy = 0;
+    D_801379D0.vz = 0x100;
+    D_80137E48 = AllocPrimitives(1U, 4);
+    prim = &g_PrimBuf[D_80137E48];
+    i = 0;
+
+    for (i = 0; i < 4; i++) {
```

Comment:
If you want to make the code a bit nicer, you can put a `const int PrimCount = 4` at the top and use it on both the for loop and as a second parameter to `AllocPrimitives`